repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/resources/META-INF/schema/config/openstack-identity-v3.xsd

```diff
@@ -317,6 +313,71 @@
         </xs:attribute>
     </xs:complexType>
 
+    <xs:complexType name="CacheType">
+        <xs:sequence>
+            <xs:element name="timeouts" type="CacheTimeoutsType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="atom-feed" type="AtomFeedType" minOccurs="0" maxOccurs="unbounded"/>
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="CacheTimeoutsType">
+        <xs:annotation>
+            <xs:documentation>
+                <!-- todo: html:p>
+                    The number of seconds which cached data will live in the datastore. Different data is cached
+                    separately, so there are multiple configurable cache timeouts. Each timeout value behaves in the
+                    following way:
+                    If -1, caching is disabled.
+                    If 0, data is cached indefinitely. In other words, data is eternal.
+                    If greater than 0, data is cached for the value provided, in seconds.
+                </html:p-->
+                <html:p>
+                    The number of seconds which cached data will live in the datastore. Different data is cached
+                    separately, so there are multiple configurable cache timeouts. Each timeout value behaves in the
+                    following way:
+                    If 0, data is cached indefinitely. In other words, data is eternal.
+                    If greater than 0, data is cached for the value provided, in seconds.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:all>
+            <xs:element name="token" type="ZeroOrPositiveInteger" default="600" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="group" type="ZeroOrPositiveInteger" default="600" minOccurs="0" maxOccurs="1"/>
+            <!-- todo: xs:element name="user" type="ZeroOrPositiveInteger" default="600" minOccurs="0" maxOccurs="1"/-->
+        </xs:all>
+
+        <xs:attribute name="variance" type="ZeroOrPositiveInteger" default="0" use="optional">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        Cache timeout variance (in milliseconds) for cached data. A random value
```

Comment:
This variance should only be applied if we are going to the set value, if expiration is less we should probably not be doing the variance. I haven't looked down at the code, but ti was something that just occured to me.